sycamore/transforms/merge_elements.py

```diff
@@ -0,0 +1,130 @@
+        pass
+
+class GreedyElementMerger(ElementMerger):
+
+    def __init__(
+        self,
+        tokenizer,
+        max_tokens
+    ):
+        self.tokenizer = tokenizer
+        self.max_tokens = max_tokens
+
+
+    def _should_merge(self, element1: Tuple[Element, int], element2: Tuple[Element, int]) -> bool:
+        if element2[0].type == 'Title':
+            return False
+        if element1[1] + 1 + element2[1] > self.max_tokens:
+            return False
+        return True
+
+    def _merge(self, element1: Tuple[Element, int], element2: Tuple[Element, int]) -> Tuple[Element, int]:
+        """
+        Merge two elements; the new element's fields will be set as:
+        
+            type: "Section"
+            binary_representation: elt1.binary_representation + elt2.binary_representation
+            text_representation: elt1.text_representation + '\n' + elt2.text_representation
+            bbox: the minimal bbox that contains both elt1's and elt2's bboxes
+            properties: elt1's properties + any of elt2's properties that are not in elt1
+                > note: if elt1 and elt2 have different values for the same property, we take elt1's value
+            
+            > note: if any input field is None we take the other element's field without merge logic
+
+        Args:
+            element1 (Tuple[Element, int]): the first element (and number of tokens in it)
+            element2 (Tuple[Element, int]): the second element (and number of tokens in it)
+
+        Returns:
+            Tuple[Element, int]: a new merged element from the inputs (and number of tokens in it)
+        """
+        elt1, tok1 = element1
+        elt2, tok2 = element2
+        new_toks = 0
+        new_elt = Element()
+        new_elt.type = "Section"
+        # Merge binary representations by concatenation
+        if elt1.binary_representation == None or elt2.binary_representation == None:
+            new_elt.binary_representation = elt1.binary_representation or elt2.binary_representation
+        else:
+            new_elt.binary_representation = elt1.binary_representation + elt2.binary_representation
+        # Merge text representations by concatenation with a newline
+        if elt1.text_representation == None or elt2.text_representation == None:
+            new_elt.text_representation = elt1.text_representation or elt2.text_representation
+            new_toks = max(tok1, tok2)
+        else:
+            new_elt.text_representation = elt1.text_representation + '\n' + elt2.text_representation
+            new_toks = tok1 + 1 + tok2
+        # Merge bbox by taking the coords that make the largest box
+        if elt1.bbox == None and elt2.bbox == None:
+            pass
+        elif elt1.bbox == None or elt2.bbox == None:
+            new_elt.bbox = elt1.bbox or elt2.bbox
+        else:
+            new_elt.bbox = BoundingBox(
+                min(elt1.bbox.x1, elt2.bbox.x1),
```

Comment:
can't this result in the bounding box include other elements? seems to assume that boxes are ordered visually

Reply:
can totally result in overlapping boxes; I'm not convinced that's a huge issue though - maybe cold be a feature wherein you go back and forth with text merging and geometric merging until your elements are stable and large? The big assumption that this makes is that the elements are on the same page - not sure how we should be handling cases where that isn't true.

Reply:
not only bounding box, but also text. Are the ordering across multiple columns handled here(Roland do some work here)? In summary, is there any concern to have this as a standalone transformation without ordering?

Reply:
multi-column is not a part of this transform, no. I think there's a different transform that does that? But yes, I do assume some ordering of the elements

Reply:
Bounding boxes already overlap in some cases, so not too worried about that.